tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Hierarchical.cpp

```diff
@@ -0,0 +1,2383 @@
+            package_data0.volume_data)) == result_tensor1);
+}
+
+void two_neighbors() noexcept {
+  constexpr size_t dim = 1;
+  const size_t order = 3;
+  const Mesh<dim> mesh(order + 1, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto);
+  const size_t num_pts = mesh.number_of_grid_points();
+  const auto x = logical_coordinates(mesh);
+  using Limiter =
+      Hierarchical<dim, tmpl::list<ScalarTag<0>, VectorTag<dim, 0>>>;
+  // Use non-unity (because that's the default) but close alpha values to make
+  // the math easier but still test thoroughly.
+  Limiter hierarchical{
+      HierarchicalType::Krivodonova,
+      std::vector<double>(
+          Spectral::maximum_number_of_points<Spectral::Basis::Legendre>, 0.99)};
+
+  NeighborData<dim, typename Limiter::PackagedData> neighbor_data{};
+
+  const Element<dim> element(ElementId<dim>{0}, {});
+  // We don't care about the ElementId for these tests, just the direction.
+  Limiter::PackagedData& package_data_upper = neighbor_data[std::make_pair(
+      Direction<dim>::upper_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lower = neighbor_data[std::make_pair(
+      Direction<dim>::lower_xi(), ElementId<dim>{0})];
+
+  package_data_upper.volume_data.initialize(num_pts);
+  package_data_upper.mesh = mesh;
+  package_data_lower.volume_data.initialize(num_pts);
+  package_data_lower.mesh = mesh;
+
+  Scalar<DataVector> self_lnc(num_pts, 0.0);
+  tnsr::I<DataVector, dim> self_vector(num_pts, 0.0);
+  DataVector expected(num_pts);
+  const auto helper =
+      [
+        &element, &expected, &hierarchical, &mesh, &neighbor_data,
+        &package_data_lower, &package_data_upper, &self_lnc, &self_vector, &x
+      ](const ModalVector& upper_coeffs,
+        const std::vector<double>& initial_coeffs,
+        const ModalVector& lower_coeffs,
+        const std::vector<double>& expected_coeffs) noexcept {
+    set_legendre(&get(self_lnc), mesh, x, initial_coeffs);
+    get(get<::Tags::Modal<ScalarTag<0>>>(package_data_upper.volume_data)) =
+        upper_coeffs;
+    get(get<::Tags::Modal<ScalarTag<0>>>(package_data_lower.volume_data)) =
+        lower_coeffs;
+    for (size_t i = 0; i < dim; ++i) {
+      set_legendre(&self_vector.get(i), mesh, x, initial_coeffs);
+      get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_upper.volume_data)
+          .get(i) = upper_coeffs;
+      get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_lower.volume_data)
+          .get(i) = lower_coeffs;
+    }
+    hierarchical(&self_lnc, &self_vector, element, mesh, neighbor_data);
+    set_legendre(&expected, mesh, x, expected_coeffs);
+    CHECK_ITERABLE_APPROX(get(self_lnc), expected);
+    for (size_t i = 0; i < dim; ++i) {
+      CHECK_ITERABLE_APPROX(self_vector.get(i), expected);
+    }
+  };
+  // Note: Throughout the tests 1.0e-18 is used as a +0.
+
+  // Limit no coefficients because the highest coefficient doesn't need
+  // limiting.
```

Comment:
Please add a comment giving the thresholds (i.e., what is the magical value that you are tweaking coeffs to be above/below, and how is this magical value optained?), so the test is easier to parse and/or modify in the future.

Reply:
There is no magic value, it's computed by the minmod function from the finite difference. It's given in detail in the documentation for a 3d element. I'm not sure it would be enlightening to copy-paste the documentation here.